source/iot/MqttRequestResponseClient.cpp

```diff
@@ -0,0 +1,493 @@
+
+                struct aws_mqtt_rr_client_operation *stream =
+                    aws_mqtt_request_response_client_create_streaming_operation(client, &streamingOptions);
+                if (nullptr == stream)
+                {
+                    Aws::Crt::Delete(implHandle, allocator);
+                    return nullptr;
+                }
+
+                auto impl = Aws::Crt::MakeShared<StreamingOperationImpl>(
+                    allocator, stream, options, aws_mqtt_request_response_client_get_event_loop(client));
+                auto streamingOperation = Aws::Crt::MakeShared<StreamingOperation>(allocator, impl);
+
+                implHandle->m_allocator = allocator;
+                implHandle->m_impl = impl;
+
+                return streamingOperation;
+            }
+
+            StreamingOperation::~StreamingOperation()
+            {
+                m_impl->Close();
+            }
+
+            void StreamingOperation::Open()
+            {
+                m_impl->Open();
+            }
+
+            //////////////////////////////////////////////////////////
+
+            struct IncompleteRequest
+            {
+                IncompleteRequest() : m_allocator(nullptr), m_handler() {}
+
+                struct aws_allocator *m_allocator;
+
+                UnmodeledResultHandler m_handler;
+            };
+
+            static void s_completeRequestWithError(struct IncompleteRequest *incompleteRequest, int errorCode)
+            {
+                UnmodeledResult result(errorCode);
+                incompleteRequest->m_handler(std::move(result));
+            }
+
+            static void s_completeRequestWithSuccess(
+                struct IncompleteRequest *incompleteRequest,
+                const struct aws_byte_cursor *response_topic,
+                const struct aws_byte_cursor *payload)
+            {
+                UnmodeledResponse response;
+                response.WithTopic(*response_topic);
+                response.WithPayload(*payload);
+
+                UnmodeledResult result(response);
+                incompleteRequest->m_handler(std::move(result));
+            }
+
+            static void s_onRequestComplete(
+                const struct aws_byte_cursor *response_topic,
+                const struct aws_byte_cursor *payload,
+                int error_code,
+                void *user_data)
+            {
+                auto *incompleteRequest = static_cast<IncompleteRequest *>(user_data);
+
+                if (error_code != AWS_ERROR_SUCCESS)
+                {
+                    s_completeRequestWithError(incompleteRequest, error_code);
+                }
+                else
+                {
+                    s_completeRequestWithSuccess(incompleteRequest, response_topic, payload);
+                }
+
+                Aws::Crt::Delete(incompleteRequest, incompleteRequest->m_allocator);
+            }
+
+            class MqttRequestResponseClientImpl
```

Comment:
I'm not super clear what is purpose of `MqttRequestResponseClientImpl`. So far the `Impl` worked as a "MqttRequestResponseClientCore" which handles the releasing of the client. Meanwhile, from the name, I got the idea that we probably wanna extend the `MqttRequestResponseClient` with different client implementation. Would we have plan for different client implementation?

Reply:
There are no plans for different implementations.  I just prefer "Impl" to "Core"